apps/server/src/modules/tool/service/validation/external-tool-validation.service.ts

```diff
@@ -14,11 +14,14 @@ export class ExternalToolValidationService {
 	async validateCreate(externalToolDO: ExternalToolDO): Promise<void> {
 		await this.commonToolValidationService.validateCommon(externalToolDO);
 
-		if (
-			this.externalToolService.isOauth2Config(externalToolDO.config) &&
-			!(await this.isClientIdUnique(externalToolDO))
-		) {
-			throw new ValidationError(`The Client Id of the tool ${externalToolDO.name} is already used.`);
+		if (this.externalToolService.isOauth2Config(externalToolDO.config)) {
```

Comment:
...else do nothing for this context sounds wrong

Reply:
hm... this is a validation method so else do not throw does seem fine.... we are calling this from a use case, where  we call validate and then call create... what would you prefer here?

Reply:
discussed